packages/server/src/services/apikey/index.ts

```diff
@@ -1,25 +1,87 @@
 import { StatusCodes } from 'http-status-codes'
-import { addAPIKey, deleteAPIKey, getAPIKeys, updateAPIKey } from '../../utils/apiKey'
+import {
+    addAPIKey as addAPIKey_json,
+    deleteAPIKey as deleteAPIKey_json,
+    generateAPIKey,
+    generateSecretHash,
+    getApiKey as getApiKey_json,
+    getAPIKeys as getAPIKeys_json,
+    updateAPIKey as updateAPIKey_json,
+    replaceAllAPIKeys as replaceAllAPIKeys_json,
+    importKeys as importKeys_json
+} from '../../utils/apiKey'
 import { addChatflowsCount } from '../../utils/addChatflowsCount'
-import { getApiKey } from '../../utils/apiKey'
 import { InternalFlowiseError } from '../../errors/internalFlowiseError'
 import { getErrorMessage } from '../../errors/utils'
+import { getRunningExpressApp } from '../../utils/getRunningExpressApp'
+import { ApiKey } from '../../database/entities/ApiKey'
+import { APIKEYS_STORAGE_DB, APIKEYS_STORAGE_JSON } from '../../AppConfig'
+import { randomBytes } from 'crypto'
+import { Not, IsNull } from 'typeorm'
 
 const getAllApiKeys = async () => {
     try {
-        const keys = await getAPIKeys()
-        const dbResponse = await addChatflowsCount(keys)
-        return dbResponse
+        if (APIKEYS_STORAGE_JSON) {
```

Comment:
so if user that is using existing version, and upgrade to new version which includes this change, they wont have the storage type set right? from the code this goes into else block, and throw an error.

should we do something similar like Yau was doing. If no key is set initially, we migrate the existing apikey.json to database, then the database approach will be the default unless specified otherwise

Reply:
for the first time user, the key is set to APIKEYS_STORAGE_JSON. This is true for when the value is 'json' or when the key is missing. [AppConfig.ts - Line 1](https://github.com/FlowiseAI/Flowise/pull/2783/files#diff-cb7edb0c2cb215e1fc2456c1cc8ee8c7de92f29ba9d9e9f53d4efd1989f4c224R1)